groups/tests/test_forms.py

```diff
@@ -88,17 +139,43 @@ def get_form(self, **kwargs):
 
     def test_form_valid(self):
         form = self.get_form(data={})
-
         self.assertTrue(form.is_valid(), msg=form.errors)
 
     def test_initial_subscribe(self):
         form = self.get_form()
-
         self.assertTrue(form.initial['subscribe'])
 
     def test_initial_unsubscribe(self):
         self.discussion.subscribers.add(self.user)
-
         form = self.get_form()
-
         self.assertFalse(form.initial['subscribe'])
+
+    def test_submit_not_input(self):
+        """The form does not have a submit <input>."""
+        rendered_form = render_crispy_form(self.get_form())
+        inputs = BeautifulSoup(rendered_form, 'html.parser').findAll('input')
+        has_submit = any(i for i in inputs if i.get('type') == 'submit')
+        self.assertFalse(has_submit)
+
+    def test_submit_button(self):
+        """
+        The form has a submit <button>.
+
+        This allows for more flexibility in styling.
+        """
+        rendered_form = render_crispy_form(self.get_form())
+        button = BeautifulSoup(rendered_form, 'html.parser').findAll('button')[0]
+        self.assertEqual(button.get('type'), 'submit')
+
+    def test_button_text_subscribe(self):
+        """The button says 'Subscribe' when the user is not subscribed."""
+        rendered_form = render_crispy_form(self.get_form())
+        button = BeautifulSoup(rendered_form, 'html.parser').findAll('button')[0]
+        self.assertEqual(button.string, 'Subscribe')
+
+    def test_button_text_unsubscribe(self):
+        """The button says 'Unsubscribe' when the user is subscribed."""
+        self.discussion.subscribers.add(self.user)
+        rendered_form = render_crispy_form(self.get_form())
+        button = BeautifulSoup(rendered_form, 'html.parser').findAll('button')[0]
```

Comment:
Is there a `find('button')` method that'll do the same thing as `findAll('button')[0]`?